src/pathOrLazy.js

```diff
@@ -0,0 +1,26 @@
+import { curryN, isNil, nAry, pathOr, when } from 'ramda';
+
+/**
+ * If the given, non-null object has a value at the given path, returns the value at that path.
+ * Otherwise returns the result of invoking the provided function.
+ *
+ * @func pathOrLazy
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.22.0|v2.22.0}
+ * @category Object
+ * @typedefn Idx = String | Int
+ * @sig (() -> a) -> [Idx] -> {a} -> a
+ * @param {Function} fd The function that will return the default value.
+ * @param {Array} p The path to use.
+ * @param {Object} obj The object to retrieve the nested property from.
+ * @return {*} The data at `path` of the supplied object or the default value.
+ * @example
+ *
+ * RA.pathOrLazy(() => 'N/A', ['a', 'b'], {a: {b: 2}}); //=> 2
+ * RA.pathOrLazy(() => 'N/A', ['a', 'b'], {c: {b: 2}}); //=> "N/A"\
+ */
+const pathOrLazy = curryN(3, function pathOrLazy(fd, p, obj) {
```

Comment:
What do say to the following implementation ? It's simpler to reason about (for me sugjectively) and more explicit what is going on.
Also let's not deal with the arity problem - there is nothing we can do about it and it may be expected behavior to return function with first parameter applied by running it. Also it would be beneficial to pass the `obj`  into the default function to be able to compute the state from object in pure manner. We leave `pathOr` to define to return `defaultFn` and if it  is returns we call it with `obj`. Otherwise we return original value return by `pathOr`.
```js
const pathOrLazy = curryN(3, (defaultFn, path, obj) => when(
  identical(defaultFn), 
  partial(defaultFn, [obj]), 
  pathOr(defaultFn, path, obj)
))
```

Reply:
@char0n - thanks for the suggestion. I like your approach better :)

I updated the PR to include your suggestion but I also wrapped the `defaultFn` in `unary` so that `when` doesn't pass the `defaultFn` to itself as the second parameter after `obj`.  If you do not like this change then let me know and I can remove `unary`.  What are your thoughts?

Reply:
@willcaul fine with me, if unary is solving some case i didn't see before let's leave it as you suggest.